src/services/pwabuilder.js

```diff
@@ -36,17 +36,21 @@ PWABuilder.prototype.createManifestFromUrl = function (url, client) {
 
       var manifest = _.assign(body, { id: uuid.v4().slice(0, 8) });
 
-      self
-        .validateManifest(manifest)
-        .then(function (manifest) {
-          client.set(
-            manifest.id,
-            JSON.stringify(manifest),
-            'EX',
-            expirationTime
-          );
-          return resolve(manifest);
-        }).fail(reject);
+        try {
+          self
+            .validateManifest(manifest)
+            .then(function (manifest) {
+              client.set(
+                manifest.id,
+                JSON.stringify(manifest),
+                'EX',
+                expirationTime
+              );
+              return resolve(manifest);
+            }).fail(reject);
+        } catch (e) {
+          return reject(e);
+        }
```

Comment:
exception kills the service, so this mitigates the effects of us calling it twice